app/assets/javascripts/modules/FilterTable.js

```diff
@@ -45,34 +46,49 @@ define(['jquery', 'DoughBaseComponent', 'List'],
   };
 
   /**
-   * makeFilterTargetClasses
+   * toggleFilterWrappers function
    *
-   * Find columns with filterTargetClass and apply individual classes
-   * to them, since this is what ListJS wants.
+   * show/hide fitlerWrappers when all filterTargetClasses are hidden
    *
-   * @return {Array} array of classes used
    */
-  FilterTableProto.makeFilterTargetClasses = function () {
-    var filterClasses = [];
+  FilterTableProto.toggleFilterWrappers = function() {
+  	var self = this;
 
-    this.$el.find('.' + this.config.filterTargetClass).each(function(_, cell) {
-      var $cell = $(cell),
-          filterClass = 'js-filterable-' + $cell.index();
-      $cell.addClass(filterClass);
-      filterClasses.push(filterClass);
-    });
+		$(self.config.filterWrapperClass).each( function(index, element ) {
+			self.toggleFilterWrapper( $(element) )
+		});
+	}
 
-    return $.unique(filterClasses);
+  /**
+   * toggleFilterWrapper function
+   *
+   * show/hide fitlerWrapper when all filterTargetClasses are hidden
+   *
+   */
+
+  FilterTableProto.toggleFilterWrapper = function(element) {
+		var list = element.find(this.config.filterListClass)
+		var rows = $(list[0]).find('tr')
+		var hidden_rows = $(rows).filter(function() {
+		  return $(this).css('display') == 'none';
+		});
+
+		if(hidden_rows.length == rows.length) {
+			element.hide();
+		} else {
+			element.show();
+		}
   };
 
   FilterTableProto.bindFilterToElements = function() {
-    var firmOptions = {
-      valueNames: this.makeFilterTargetClasses(),
-      searchClass: this.config.filterFieldClass,
-      listClass: this.config.filterListClass
-    };
+  	var self = this
 
-    new List(this.$el.attr('id'), firmOptions);
+		$(this.config.filterFieldClass).fastLiveFilter(this.config.filterListClass, {
```

Comment:
If we define a `self`, I usually prefer to use `self` throughout — even where you can use `this`. It makes the code a little easier to work with.